adoc/chapters/programming_interface.adoc

```diff
@@ -5474,58 +5362,53 @@ provided when allocating memory on the host.  The range of the constructed SYCL
 [code]#buffer# is specified by the [code]#bufferRange# parameter provided.
 Zero or more properties can be provided to the constructed SYCL [code]#buffer#
 via an instance of [code]#property_list#.
+'''
 
-a@
-[source]
+.[apititle]#Construct from iterators#
+[source,role=synopsis,id=api:buffer-ctor-iterator]
 ----
-template <typename InputIterator>
-buffer(InputIterator first, InputIterator last,
-       const property_list& propList = {})
-----
-   a@ Create a new allocated 1D buffer initialized from the given elements
-      ranging from [code]#first# up to one before [code]#last#.
-      The data is copied to an intermediate memory position by the runtime.
-      Data is not written back to the same iterator set provided. However, if the [code]#buffer# has a valid non-const iterator specified via the member function [code]#set_final_data()#, data will be copied back to that iterator.
-      The constructed SYCL [code]#buffer# will use a default constructed [code]#AllocatorT# when allocating memory on the host.
-      Zero or more properties can be provided to the constructed SYCL [code]#buffer# via an instance of [code]#property_list#.
+template <class InputIterator>
```

Comment:
For these functions accepting `InputIterator`, should there be _Constraints_?  They definitely need to be iterators.  Do they need to be contiguous/random access/whatever?  Does the `value_type` need to be convertible to the buffer type?